content/articles/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/index.md

```diff
@@ -0,0 +1,111 @@
+description: In this article, we will go over reasons that would require you to redirect DNS requests and how to redirect DNS requests.
+author: daniel-muriithi
+date: 2021-10-08T00:00:00-15:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/how-to-use-a-local-dns-resolver-to-redirect-all-dns-requests/hero.jpg
+    alt: DNS
+---
+### Introduction
+A **DNS query** `also known as a DNS request` is a request for information from a user's computer to a DNS server (DNS client). To acquire the IP address associated with a domain name, a DNS request is frequently made. A DNS client requests DNS servers to retrieve the domain's IP address when visiting an environment.
+
+When setting up a home network, one can only allow access to the Unbound DNS service on OPNsense and restrict all other DNS servers from being accessed. Any unauthorized access to external DNS servers will be prohibited by this method. Only the local DNS resolver is allowed (unless the DNS requests are encrypted).
+
+However, one can take a different approach. The local DNS server can be used to route all DNS requests to alternate `external servers`. That way, you can make that everything is working properly and limit or filter DNS requests. You can easily monitor DNS requests that would otherwise go undetected if a firewall rule had blocked them with the DNS redirection feature enabled on your computer.
+
+### Table of contents
+- [Reason for redirecting DNS requests to local DNS resolver](#reason-for-redirecting-dns-requests-to-local-dns-resolver)
+- [Creating a DNS Redirection NAT Port Forward Rule](#creating-a-dns-redirection-nat-port-forward-rule)
+- [Configuration of multiple Interfaces](#configuration-of-multiple-interfaces)
+- [Testing the redirection](#testing-the-redirection)
+- [Conclusion](#conclusion)
+
+###  Reason for redirecting DNS requests to local DNS resolver
+Even though some devices (such as Smart TVs) should be automatically configured to use the local DNS server provided by the DHCP service, some devices will attempt to use Google's DNS servers or other public DNS servers (possibly to circumvent ad blockers and to gather valuable data on DNS requests made by your device). The approach described above will not work for devices that only use alternative/hard-coded DNS servers if the machine refuses to use the preset DNS servers.
+
+### Creating a DNS Redirection NAT Port Forward Rule
+Implementing a basic NAT port forward rule to reroute DNS requests from a certain network is possible. Port forwarding rules are often used to allow traffic from an outside network on the Internet to access various services housed on your network, but they can be used on internal network interfaces as well. This means that if you unintentionally write a rule that prevents access to the web interface, you won't lose access to the web interface, as you'll notice on the `Firewall: NAT: Port Forward` page.
+

```

Comment:
@Daniel-muriithi 
I'm a little lost on where these settings are to be changed from
Are we on windows, are we on Linux?
Is it the default settings application or is it a 3rd party software we need to install?

Kindly clarify this